R/zchunk_LB134.Diet_Rfao.R

```diff
@@ -0,0 +1,380 @@
+             "L100.FAO_ag_Food_t",
+             "L100.FAO_an_Food_t",
+             "L101.ag_Food_Pcal_R_C_Y",
+             FILE = "temp-data-inject/L105.an_Food_Pcal_R_C_Y",
+             "L101.Pop_thous_R_Yh",
+             "L102.pcgdp_thous90USD_Scen_R_Y"))
+  } else if(command == driver.DECLARE_OUTPUTS) {
+    return(c("L134.pcFood_kcald_R_Dmnd_Y",
+             "L134.pcFood_kcald_R_Dmnd_Y_ssp1",
+             "L134.pcFood_kcald_R_Dmnd_Y_ssp2",
+             "L134.pcFood_kcald_R_Dmnd_Y_ssp3",
+             "L134.pcFood_kcald_R_Dmnd_Y_ssp4",
+             "L134.pcFood_kcald_R_Dmnd_Y_ssp5"))
+  } else if(command == driver.MAKE) {
+
+    all_data <- list(...)[[1]]
+
+    # Load required inputs
+    A_FoodDemand_SSPs <- get_data(all_data, "aglu/A_FoodDemand_SSPs")
+    iso_GCAM_regID <- get_data(all_data, "common/iso_GCAM_regID")
+    AGLU_ctry <- get_data(all_data, "aglu/AGLU_ctry")
+    FAO2050_items_cal <- get_data(all_data, "aglu/FAO2050_items_cal")
+    get_data(all_data, "aglu/FAO2050_Diet")  %>%
+      gather(year, value, -FAO2050_reg, -FAO2050_item) %>%
+      mutate(year = as.integer(year)) ->
+      FAO2050_Diet
+    L100.FAO_ag_Food_t <- get_data(all_data, "L100.FAO_ag_Food_t")
+    L100.FAO_an_Food_t <- get_data(all_data, "L100.FAO_an_Food_t")
+    L101.ag_Food_Pcal_R_C_Y <- get_data(all_data, "L101.ag_Food_Pcal_R_C_Y")
+    L105.an_Food_Pcal_R_C_Y <- get_data(all_data, "temp-data-inject/L105.an_Food_Pcal_R_C_Y") %>%
+      # TEMPORARY - for temp-data-inject data
+      gather(year, value, -GCAM_region_ID, -GCAM_commodity) %>% mutate(year = as.integer(substr(year, 2, 5)))
+    L101.Pop_thous_R_Yh <- get_data(all_data, "L101.Pop_thous_R_Yh")
+    L102.pcgdp_thous90USD_Scen_R_Y <- get_data(all_data, "L102.pcgdp_thous90USD_Scen_R_Y")
+
+    # Build historical time series of per-capita caloric demands
+
+    # Historical time series of ag and animal product consumption
+    # Original lines 41-49
+    L101.ag_Food_Pcal_R_C_Y %>%
+      filter(year %in% AGLU_HISTORICAL_YEARS) %>%
+      group_by(GCAM_region_ID, year) %>%
+      summarise(value = sum(value)) %>%
+      mutate(GCAM_demand = "crops") ->
+      L134.ag_Food_Pcal_R_Y
+
+    L105.an_Food_Pcal_R_C_Y %>%
+      filter(year %in% AGLU_HISTORICAL_YEARS) %>%
+      group_by(GCAM_region_ID, year) %>%
+      summarise(value = sum(value)) %>%
+      mutate(GCAM_demand = "meat") %>%
+      bind_rows(L134.ag_Food_Pcal_R_Y) ->
+      L134.Food_Pcal_R_Dmnd_Y
+
+    # Divide by population to calculate the historical per-capita food demands, in kcal per person per day
+    # Original lines 51-56
+    L134.Food_Pcal_R_Dmnd_Y %>%
+      left_join_error_no_match(L101.Pop_thous_R_Yh, by = c("GCAM_region_ID", "year")) %>%
+      mutate(value = value.x * CONV_DAYS_YEAR * (1 / CONV_MCAL_PCAL ) / value.y) %>%
+      select(-value.x, -value.y) ->
+      L134.pcFood_kcald_R_Dmnd_Y
+
+    # Extrapolate this to the future periods based on the FAO projections
+    # Original lines 58-69
```

Comment:
It's not clear to me how this pair of pipelines extrapolates to future years. It seems like we just pick the last historical year and sum things up by iso in that year.